Migrate to Yarn v3

Gudahtt opened the pull request:
Yarn has been updated following the steps taken in the module template
repository: https://github.com/MetaMask/metamask-module-template/pull/108

`auto-changelog` has also been updated; the version used previously was not compatible with Yarn v3.

shanejonas commented on tsconfig.json:
```diff
@@ -1,14 +1,16 @@
+    "target": "ES2017"
   },
   "include": ["src/**/*.ts"],
-  "exclude": ["node_modules", "types"]
```
should we keep exclude node_modules? or atleast the module template should probably have that

Gudahtt replied:
It's already excluded by default. [From the docs](https://www.typescriptlang.org/tsconfig#exclude):

>Default: `["node_modules", "bower_components", "jspm_packages"]`, plus the value of `outDir` if one is specified.

Gudahtt commented on tsconfig.json:
```diff
@@ -1,14 +1,16 @@
     "declaration": true,
+    "esModuleInterop": true,
+    "inlineSources": true,
+    "lib": ["ES2020"],
```
It looks like this broke the tests. I think I need to include `DOM` here.

Gudahtt replied:
Addressed in f4b28dcab34066af1d68e389a59eb816582b087d

rekmarks reviewed (approved):
LGTM!

Gudahtt committed:
Migrate to Yarn v3

Yarn has been updated following the steps taken in the module template
repository: https://github.com/MetaMask/metamask-module-template/pull/108

`auto-changelog` has also been updated; the version used previously was
not compatible with Yarn v3.

Gudahtt commented:
This doesn't depend on any other PRs, but I'd recommend trying to land https://github.com/MetaMask/browser-passworder/pull/23 first so that we get tests running again.

darkwing reviewed (approved):
Works great!  Thank you @Gudahtt !
